sdk/containers/azcontainerregistry/digest_helper.go

```diff
@@ -0,0 +1,160 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azcontainerregistry
+
+import (
+	"crypto/sha256"
+	"encoding"
+	"errors"
+	"fmt"
+	"hash"
+	"io"
+	"strings"
+)
+
+var (
+	validatorCtors           = map[string]func() digestValidator{"sha256": newSha256Validator}
+	ErrMismatchedHash        = errors.New("mismatched hash")
+	ErrDigestAlgNotSupported = errors.New("digest algorithm not supported")
+)
+
+type digestValidator interface {
+	io.Writer
+	validate(digest string) error
+}
+
+func parseDigestValidator(digest string) (digestValidator, error) {
+	alg := digest[:strings.Index(digest, ":")]
+	if v, ok := validatorCtors[alg]; ok {
+		return v(), nil
+	} else {
+		return nil, ErrDigestAlgNotSupported
+	}
```

Comment:
In general, you may want to read https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow to remove the `else`.

Reply:
It applies to all occurrences in this file.

Reply:
Will fix before GA.